lib/compat/wordpress-6.8/class-gutenberg-rest-post-counts-controller.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Post_Counts_Controller class
+ *
+ * @since 6.8.0
+ *
+ * @package gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to return post counts by post type via the REST API.
+ *
+ * @since 6.8.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Post_Counts_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'counts';
```

Comment:
Hmm, I'm not sure that I love an endpoint specific for post counting. I am not against potentially adding this capability, but I think it feels more natural to be available under the `wp/v2/<post_type>/count` endpoint.

I am not sure if there's a technical reason why that would be difficult. But wondering if that could support retrieving the overall count for a specific post type (no arguments) OR with any other passed specific query arguments since they should already be supported within that post type's controller.

@TimothyBJacobs Do you have any opinion on any of these approaches?

Reply:
I've thrown up an alternative that adds a `/count` route to `WP_REST_Posts_Controller`, so `wp/v2/<post_type>/count` as @desrosj mentions above. 

- https://github.com/WordPress/gutenberg/pull/67719

I'm feeling more comfortable with that version 😄 

For Gutenberg, it's only needed for the `page` post type, but in the Core sync it'd be available for all post types that inherit from the controller.

Reply:
> I'm feeling more comfortable with that version 😄

I withdraw that position for the reasons stated in:

https://github.com/WordPress/gutenberg/pull/66294#issuecomment-2530040206